4_cv_basics/4_convolutions_filtering/4_convolutions_filtering.cpp

```diff
@@ -0,0 +1,113 @@
+#include <opencv2/opencv.hpp>
+#include <iostream>
+
+/*
+    Specifying Namespaces
+    Namespace is a declarative region that provides a scope to the identifiers (the names of types, functions, variables, etc) inside it. 
+*/
+using namespace std;
+using namespace cv;
+
+/// @brief : Convolution will help apply different kernels to your images yielding different results
+/// @param   Original_image
+/// @param   kernel
+/// @return : An Output Image with the kernel applied
+Mat convolve(Mat kernel, Mat original_image){
+    
+    Mat kernel_inv{kernel.size(), kernel.type()}, temp_kernel{kernel.size(), kernel.type()};
+    Mat resultant_image{original_image.size(), original_image.type()};
```

Comment:
As no padding is applied(valid convolution), the `resultant_image` should be of shape (M-a+1, N-b+1) and not of (M, N) where (M,N) is the shape of `original_image` and (a,b) is the shape of `kernel`.